### PR TITLE
Set should have a difference function

### PR DESCRIPTION
**Pull request by derekwaynecarr**
I have a need to diff sets.  It's useful to have a diff function.

@smarterclayton

### CODE REVIEW

**Comment by smarterclayton** [comment]
LGTM

File: pkg/util/set.go [hunk 1]
```diff
@@ -62,6 +62,17 @@ func (s StringSet) HasAll(items ...string) bool {
 	return true
 }
 
+// Difference returns a set of objects that are not in s2
+func (s StringSet) Difference(s2 StringSet) StringSet {
```
Comment (thockin) [inline pkg/util/set.go]: For non mathematicians (me) who had to think hard about this, can you make the comment a bit more explicit?  Maybe a small example?

Comment (derekwaynecarr) [inline pkg/util/set.go]: sure

**Comment by derekwaynecarr** [comment]
Example added inline comment:
// Difference returns a set of objects that are not in s2
// For example:
// s1 = {1, 2, 3}
// s2 = {1, 2, 4, 5}
// s1.Difference(s2) = {3}

File: pkg/util/set_test.go [hunk 2]
```diff
@@ -98,3 +98,22 @@ func TestStringSetList(t *testing.T) {
...
+	if len(c) != 1 {
+		t.Errorf("Expected len=1: %d", len(c))
+	}
+	if !c.Has("3") {
+		t.Errorf("Unexpected contents: %#v", c.List())
+	}
+	if len(d) != 2 {
+		t.Errorf("Expected len=2: %d", len(d))
+	}
+	if !d.Has("4") || !d.Has("5") {
```
Comment (thockin) [inline pkg/util/set_test.go]: puke - we should probably just add an AsSlice() method.

**Pull request merged**